internal/validations/clusterlogforwarder/validate_clusterlogforwarder_service_account_test.go

```diff
@@ -0,0 +1,159 @@
+			serviceAccount, err := getServiceAccount(customClf.Spec.ServiceAccount, customClf.Namespace, k8sClient)
+			Expect(serviceAccount).To(BeNil())
+			Expect(err).To(MatchError(MatchRegexp("service account: .+ not found")))
+		})
+
+		It("should succeed when service account is found", func() {
+			providedSAName := "test-serviceAccount"
+			clfServiceAccount := &corev1.ServiceAccount{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      providedSAName,
+					Namespace: customClf.Namespace,
+				},
+			}
+			customClf.Spec.ServiceAccount = providedSAName
+			Expect(k8sClient.Create(context.TODO(), clfServiceAccount)).To(Succeed())
+
+			serviceAccount, err := getServiceAccount(customClf.Spec.ServiceAccount, customClf.Namespace, k8sClient)
+			Expect(serviceAccount).ToNot(BeNil())
+			Expect(err).To(BeNil())
+		})
+	})
+
+	Context("validate permissions", func() {
+		Context("gather clf inputs", func() {
+			var expectedSet sets.String
+
+			var clf = loggingv1.ClusterLogForwarder{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      "custom-clf",
+					Namespace: constants.OpenshiftNS,
+				},
+			}
+			BeforeEach(func() {
+				clf.Spec = loggingv1.ClusterLogForwarderSpec{}
+			})
+
+			It("should gather all inputs from clf.spec.pipelines", func() {
+				clf.Spec = loggingv1.ClusterLogForwarderSpec{
+					Pipelines: []loggingv1.PipelineSpec{
+						{
+							Name: "pipeline1",
+							InputRefs: []string{
+								loggingv1.InputNameAudit,
+								loggingv1.InputNameApplication,
+							},
+						},
+						{
+							Name: "pipeline2",
+							InputRefs: []string{
+								loggingv1.InputNameApplication,
+							},
+						},
+					},
+				}
+				expectedSet = sets.NewString(loggingv1.InputNameAudit, loggingv1.InputNameApplication)
+				inputs := gatherPipelineInputs(clf)
+				Expect(inputs).To(HaveLen(2))
+				Expect(inputs).To(Equal(expectedSet))
+			})
+		})
+		Context("subjectAccessReview", func() {
+			var (
+				client = &mockOpenShiftClient{
+					sarResponses: map[string]bool{
+						"collect-application-logs":    true,
+						"collect-infrastructure-logs": true,
+					},
+					subjectAccessErr: errors.New("error in subject review")}
+				clfServiceAccount = &corev1.ServiceAccount{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "test-serviceAccount",
+						Namespace: constants.WatchNamespace,
+					},
+				}
+			)
+
+			It("should pass validation when service account can collect specified inputs", func() {
+				Expect(validateServiceAccountPermissions(k8sClient, client, sets.NewString("application", "infrastructure"), clfServiceAccount, constants.WatchNamespace)).To(Succeed())
+			})
+
+			It("should return validation error if service account cannot collect specified inputs", func() {
+				Expect(validateServiceAccountPermissions(k8sClient, client, sets.NewString("application", "audit"), clfServiceAccount, constants.WatchNamespace)).ToNot(Succeed())
+			})
+		})
+	})
+
+	It("should not validate clusterlogforwarder named 'instance'", func() {
```

Comment:
Maybe we should add a validation that rejects CLF(openshift-logging/instance) where the SA is defined since we provide one.... or there should be "migration" that defaults this field to our legacy name.  The migration may/should allow for a consistent reconciliation flow so we don't need to treat legacy as a special case

Reply:
We can "migrate" the SA name to the legacy name for that field but I think we still need to skip validation here because we don't create the SA until we start reconciling. The first part in validating the SA is checking if it exists and for the legacy clf, the SA is non-existent until after validation (i.e, it would fail).